api/auth/approle/go.mod

```diff
@@ -2,4 +2,7 @@ module github.com/hashicorp/vault/api/auth/approle
 
 go 1.16
 
-require github.com/hashicorp/vault/api v1.12.0
+require (
```

Comment:
Would it be possible to make this require block list out all of the indirect dependencies? It would help a lot for future maintainability

Reply:
Same comment for the other go.mods :)

Reply:
Yup, will do!
I tried running `go mod tidy` but I do not see go mod files changed. I ran `go get -u ./...` which changed the go.mod file to list all indirect dependencies but it also upgraded them. The command `go get ./...` which is supposed to just list the indirect dependencies without upgrading them is making no changes inside the go mod files. Are these the commands you were also thinking of to list them?

Reply:
There's some colour here: https://go.dev/ref/mod#go-mod-file-require

If we need to list them out one at a time, that'd still be a good thing to do -- but there's probably a better way (regenerating it from scratch?)

Reply:
Taking a look now, thanks!

Reply:
I think I found the way Violet! It requires go mod version to be at least 17 but i think it is fine? Since this is the wrong repo we are making changes against, I will create a new repo on vault enterprise repo and tag you there!

Reply:
Sounds good! And yeah, that works great :)